src/api/app/controllers/trigger/errors.rb

```diff
@@ -1,4 +1,8 @@
 module Trigger::Errors
+  class NotEnabledToken < APIError
+    setup 'bad_request', 400, 'This token is not enabled.'
```

Comment:
I'd rather raise a [`403 Forbidden`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403) or even better a [`405 Method not allowed`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405) to be picky

Reply:
I'm with @ncounter about the `403`

Reply:
> I'd rather return a [`403 Forbidden`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403)

 decided to return 400 instead because I wanted to differentiate between the errors we receive, so we don't mix the original 403 Forbidden responses with the ones we introduce with not enabled tokens.

> or even better a [`405 Method not allowed`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405) to be picky

This response fit less with the fact that a token is not enabled, I would say.

Reply:
> decided to return 400 instead because I wanted to differentiate between the errors we receive, so we don't mix the original 403 Forbidden responses with the ones we introduce with not enabled tokens.

Hm, 400 is a client error. This implies the client is doing something wrong, and in this use case the client can't do anything to fix calling a disabled token. 403 is still the valid one IMO.

Reply:
> Hm, 400 is a client error. This implies the client is doing something wrong, and in this use case the client can't do anything to fix calling a disabled token. 403 is still the valid one IMO.

In this case, we could need some action both in the client or in the server:
- in the client, we could change the authorization credentials
- in the server, we should enable the token

I went with a 400 because of this sentence if found [here](https://http.dev/400):

_There is some ambiguity between 4XX codes and there are some cases that are not explicitly covered. In situations such as these, the server may return the 400 Bad Request status as a catch-all approach._

Reply:
> _There is some ambiguity between 4XX codes and there are some cases that are not explicitly covered. In situations such as these, the server may return the 400 Bad Request status as a catch-all approach._

For me there is no ambiguity, the client cannot do anything from its side to change the fact that the token is disabled, so it's not a 400 Bad Request, it's not the client's fault.

Reply:
> For me there is no ambiguity, the client cannot do anything from its side to change the fact that the token is disabled, so it's not a 400 Bad Request, it's not the client's fault.

That sounds more like we should return a error in the range of the 500's (Server error) than in the range of the 400's (Client Error)...

Reply:
I found mentions on invalid tokens here:
- https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403#examples
- https://http.dev/498 (unofficial)
- https://http.dev/499 (unofficial)

Reply:
> > For me there is no ambiguity, the client cannot do anything from its side to change the fact that the token is disabled, so it's not a 400 Bad Request, it's not the client's fault.
> 
> That sounds more like we should return an error in the range of the 500's (Server error) than in the range of the 400's (Client Error)...

Nope, :sweat_smile: 500 is something unexpected. We know why we don't let the user run this token, it's because it's disabled. It's not unexpected and it's not the client's fault. We are forbidding the use of this token for whatever reason, so it should be a 403 Forbidden! :smile:

Reply:
I changed it to a 403 error.